internal/otelcollector/config/metric/gateway/service_test.go

```diff
@@ -45,6 +45,7 @@ func TestService(t *testing.T) {
 				"filter/drop-if-input-source-runtime",
 				"filter/drop-if-input-source-prometheus",
 				"filter/drop-if-input-source-istio",
+				"filter/drop-envoy-metrics-if-disabled",
```

Comment:
Is the filter only enabled if the istio input is enabled? It is listed in so many test cases.. looks suspicious

Reply:
This, at the moment, is how config generation is implemented. Currently, this config generation section of an individual pipeline instance is not aware of other (if any exist) pipeline instances. We need to refactor the code to avoid any unnecessary component initialization, in some cases, even a single pipeline exists, and some processors are initialized and configured which in this case is not required.


I will create a follow-up task for the refactoring.